staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -140,3 +141,39 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx context.Context, obj runtime.
 	}
 	return true, false, nil
 }
+
+// AdmissionToValidateObjectDeleteFunc returns a admission validate func for object deletion
+func AdmissionToValidateObjectDeleteFunc(admit admission.Interface, staticAttributes admission.Attributes, objInterfaces admission.ObjectInterfaces) ValidateObjectFunc {
+	mutatingAdmission, isMutatingAdmission := admit.(admission.MutationInterface)
+	validatingAdmission, isValidatingAdmission := admit.(admission.ValidationInterface)
+
```

Comment:
could hoist the `mutatingAdmission.Handles(admission.Delete)` check outside and short-circuit the expensive attribute building if it doesn't handle delete